### ISSUE TITLE
ClassCastException when using @AuthenticationPrincipal 

### ISSUE THREAD

**Issue by straumat** [theborakompanioni/bitcoin-spring-boot-starter#122]
I have a spring boot controller with that code:
```
@GetMapping("/account/settings")
@PreAuthorize("isAuthenticated()")
public String displayAccountSettings(final Model model,
                                         final @AuthenticationPrincipal(errorOnInvalidType = true) UserDetails currentUser) {
        Optional<UserDTO> connectedUser = userService.getUserByUsername(currentUser.getUsername());
```

I get this error: `java.lang.ClassCastException: 028acfe2a7c5444a5872968f628d8fc827c32f7b11761e094818ee10ceeee9c5eb is not assignable to interface org.springframework.security.core.userdetails.UserDetails`

It seems your starter is not passing the userDetails but the linkingkey ?

**Comment by theborakompanioni** [comment]
Fixed in #117. :raised_hands: 

Thanks @straumat!

**Comment by straumat** [comment]
It works!
